core/modules.rs

```diff
@@ -48,10 +49,13 @@ pub trait Loader: Send + Sync {
     specifier: &str,
     referrer: &str,
     is_root: bool,
-  ) -> Result<String, Self::Error>;
+  ) -> Result<ModuleSpecifier, Self::Error>;
 
-  /// Given an absolute url, load its source code.
-  fn load(&self, url: &str) -> Box<SourceCodeInfoFuture<Self::Error>>;
+  /// Given ModuleSpecifier, load its source code.
+  fn load(
+    &self,
+    module_specifier: &ModuleSpecifier,
+  ) -> Box<SourceCodeInfoFuture<Self::Error>>;
```

Comment:
👍 I think adopting URL at this layer is simplifying.

As long as core/isolate.rs doesn't depend on URL.